example/plugins/example/plugin.js

```diff
@@ -0,0 +1,18 @@
+class ExamplePlugin {
+  constructor(pod, config) {
+    this.pod = pod;
+    this.config = config;
+    this.key = 'example';
```

Comment:
Can we drop the need to have a "key" and "name" until we absolutely need it for something? If the name of the plugin class is needed in the internals of Amagaki (i.e. for logging) let's get it with:

```
const foo = new ExamplePlugin(...);
foo.constructor.name;  // `ExamplePlugin`
```

Reply:
I've removed the `key` and `name` requirement. It now relies on having a unique class name for finding existing plugins, but we can revisit it in the future if it becomes an issue of being able to identify plugins.

example output in profiling:

```
Hook: createRenderer took 0.3ms (0.00% called 43x, 6.67µs each)
ExamplePlugin hook: createRenderer took 86.80µs (0.00% called 43x, 2.02µs each)
```